### ISSUE TITLE
DOC: fix PR02 errors in docstrings

### ISSUE THREAD

**Issue by jordan-d-murphy** [pandas-dev/pandas#57111]
Pandas has a script for validating docstrings:

https://github.com/pandas-dev/pandas/blob/9008ee5810c09bc907b5fdc36fc3c1dff4a50c55/ci/code_checks.sh#L72-L194

Currently, some methods fail the PR02 check.

The task here is:

- take 2-4 methods
- run: scripts/validate_docstrings.py --format=actions --errors=PR02 `method-name`
- check if validation docstrings passes for those methods, and if it’s necessary fix the docstrings according to whatever error is reported
- remove those methods from code_checks.sh
- commit, push, open pull request

Please don't comment `take` as multiple people can work on this issue. You also don't need to ask for permission to work on this, just comment on which methods are you going to work.

If you're new contributor, please check the [contributing guide](https://pandas.pydata.org/docs/dev/development/contributing.html)

thanks @natmokval @mroeschke for the inspiration for this issue

**Comment by jordan-d-murphy** [comment]
I'll take 
- pandas.Series.interpolate     
- pandas.read_hdf      
- pandas.HDFStore.append

**Comment by jrmylow** [comment]
I'll get started with:
* [ ] pandas.describe_option
* [ ] pandas.reset_option
* [ ] pandas.get_option
* [ ] pandas.set_option

**Comment by jordan-d-murphy** [comment]
I'll take: 

- pandas.core.groupby.SeriesGroupBy.idxmax
- pandas.core.groupby.SeriesGroupBy.idxmin

**Comment by jrmylow** [comment]
Would it be helpful to add the good-first-issue label to this? Seems the previous iteration (for EX03?) had that

**Comment by jordan-d-murphy** [comment]
I think so, but I can't add the label myself. Feel free to add if you have permissions 🙂

**Comment by thiphan94** [comment]
I worked on:
 - pandas.Series.clip
 - pandas.DataFrame.clip
 - pandas.api.extensions.ExtensionArray.argsort
 -

**Comment by jordan-d-murphy** [comment]
I'll take 
- pandas.core.groupby.SeriesGroupBy.transform
- pandas.core.groupby.DataFrameGroupBy.transform

**Comment by jordan-d-murphy** [comment]
Note - a new method was added to the list 

`pandas.core.groupby.DataFrameGroupBy.corrwith`

See: 

> https://github.com/pandas-dev/pandas/pull/57109
> https://github.com/pandas-dev/pandas/issues/57158

**Comment by jordan-d-murphy** [comment]
I'll work on 

- pandas.core.window.rolling.Rolling.quantile
- pandas.core.window.expanding.Expanding.quantile

**Comment by jordan-d-murphy** [comment]
I got a fix for `pandas.io.formats.style.Styler.to_excel`

**Comment by noah-asing** [comment]
I'll take
- [ ] pandas.Series.plot
- [ ] pandas.Series.plot.bar
- [ ] pandas.Series.plot.barh
- [ ] pandas.Series.plot.line
- [ ] pandas.Series.plot.pie

**Comment by jordan-d-murphy** [comment]
got a fix for `pandas.Series.rename_axis`

**Comment by noah-asing** [comment]
"Unknown parameters" arises from parameters in the docstring that do not appear in the function signatures. 

Often, this happens because the parameter is buried in `**kwargs`. For example, in the docstring for [bar and line plots](https://github.com/pandas-dev/pandas/blob/ce50a85e2a072754601da4183a93432892defced/pandas/plotting/_core.py#L446), `color` is specified as a parameter, but the [function signature](https://github.com/pandas-dev/pandas/blob/ce50a85e2a072754601da4183a93432892defced/pandas/plotting/_core.py#L1180) does not include it.

In these cases, the only thing I can think of is to copy the default color value, adding it to the function signature (note this wouldn't be updated if `matplotlib`'s defaults update) and then override it in the function if it is found in kwargs, but I really don't like this solution.

What would be a better method?

**Comment by jordan-d-murphy** [comment]
I've noticed this too, and I've been thinking about this a lot over the last week or so, but I'm not sure yet what the best approach would be. Multiple docstrings could be resolved with the same approach.

**Comment by jrmylow** [comment]
As a suggestion, I've implemented something that might work in: https://github.com/pandas-dev/pandas/pull/57235, open to suggestions on whether this reads well to people

**Comment by jordan-d-murphy** [comment]
Cool @jrmylow! I think that has a lot of promise, I'll play around with this and report back with some more thoughts. Also, I'd love to hear what some of the core team members think of an approach like this. This could resolve quite a few of the PR02s!

**Comment by jordan-d-murphy** [comment]
I'll open a PR for `pandas.core.groupby.DataFrameGroupBy.corrwith`

**Comment by yuanx749** [comment]
Worked on `pandas.PeriodIndex`.

**Comment by jrmylow** [comment]
I'll see if I can get the tseries.offsets done in one hit

**Comment by jordan-d-murphy** [comment]
I got a fix for `pandas.core.groupby.SeriesGroupBy.apply`

**Comment by jordan-d-murphy** [comment]
I got a fix for `pandas.core.groupby.DataFrameGroupBy.hist`

**Comment by jordan-d-murphy** [comment]
I got an approach that might work for the **kwargs issue we discussed above, @noah-asing @jrmylow 

I'll open a PR for `pandas.Series.str.wrap` and let's see what some of the core team members think about an approach like this.

**Comment by jordan-d-murphy** [comment]
@Dxuian - try running this with some of the ones that haven't been fixed yet
`scripts/validate_docstrings.py --format=actions --errors=PR02 <method-name>`
for example: 
`scripts/validate_docstrings.py --format=actions --errors=PR02 pandas.Series.dt.to_period`

**Comment by jordan-d-murphy** [comment]
Ah I see, @Dxuian

Running `scripts/validate_docstrings.py --format=actions --errors=PR02 pandas.read_hdf` on my branch outputs the following, 

```
################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.read_hdf" correct. :)
```

which makes me think what might be going on is your branch might be out of sync with main. 

I've been using these commands to keep my env and branches in sync with the latest updates. Can you try updating your branch with the latest changes from main?

[Updating the development environment](https://pandas.pydata.org/docs/dev/development/contributing.html#id12)

git checkout main
git merge upstream/main
mamba activate pandas-dev
mamba env update -f environment.yml --prune


[Creating a feature branch](https://pandas.pydata.org/docs/development/contributing.html#id7)

git checkout main
git pull upstream main --ff-only
git checkout -b shiny-new-feature


[Pre-commit](https://pandas.pydata.org/docs/development/contributing_codebase.html#id3)

pre-commit install
pre-commit run --files <files you have modified>
pre-commit run --from-ref=upstream/main --to-ref=HEAD --all-files
pre-commit run --hook-stage manual --all-files



[Updating your pull request](https://pandas.pydata.org/docs/development/contributing.html#id11)

git checkout shiny-new-feature
git fetch upstream
git merge upstream/main

**Comment by jordan-d-murphy** [comment]
I got fixes for these: 
```
pandas.core.groupby.DataFrameGroupBy.rolling
pandas.core.groupby.SeriesGroupBy.rolling
```

**Comment by jordan-d-murphy** [comment]
Not sure what the general consensus is on modifying **.pyx** files, but I've found something that will silence the PR02 errors for situations where the parameters are accurately documented in the `.pyx` file, but failing the numpydoc validations with PR02 errors for `Unknown Parameters`

**Example:   pandas.tseries.offsets.DateOffset**

`scripts/validate_docstrings.py --format=actions --errors=PR02 pandas.tseries.offsets.DateOffset`

<img width="970" alt="Screenshot 2024-02-10 at 6 01 36 PM" src="https://github.com/pandas-dev/pandas/assets/35613487/2b6306bc-9c9c-45c6-8b9c-36753ff2659e">

Docs are correct as seen here: 
https://pandas.pydata.org/docs/reference/api/pandas.tseries.offsets.DateOffset.html

@datapythonista @mroeschke @rhshadrach any thoughts or opinions? this could resolve ~ 40 PR02 errors in `pandas.tseries.offsets` alone 

Opening PR for feedback:

**Comment by jordan-d-murphy** [comment]
for anyone interested, I've opened some similar issues: 
[DOC: fix RT03 errors in docstrings](https://github.com/pandas-dev/pandas/issues/57416)
[DOC: fix SA01 errors in docstrings](https://github.com/pandas-dev/pandas/issues/57417)
[DOC: fix PR07 errors in docstrings](https://github.com/pandas-dev/pandas/issues/57420)
[DOC: fix PR01 errors in docstrings](https://github.com/pandas-dev/pandas/issues/57438)
[DOC: fix ES01 errors in docstrings](https://github.com/pandas-dev/pandas/issues/57440)
[DOC: fix GL08 errors in docstrings](https://github.com/pandas-dev/pandas/issues/57443)

**Comment by pmhatre1** [comment]
Got fixes for 
pandas.DatetimeIndex.day_name
pandas.DatetimeIndex.month_name

**Comment by jordan-d-murphy** [comment]
Hi all, if https://github.com/pandas-dev/pandas/pull/57826 gets merged in, I might be reworking our approach here; this would look like closing the following issues: 

https://github.com/pandas-dev/pandas/issues/57443
https://github.com/pandas-dev/pandas/issues/57438
https://github.com/pandas-dev/pandas/issues/57420
https://github.com/pandas-dev/pandas/issues/57417
https://github.com/pandas-dev/pandas/issues/57416
https://github.com/pandas-dev/pandas/issues/57111

And opening a new issue to address these based on the new approach. 

tl;dr 

the work can still be done, but probably under a new ticket once  #57826 is merged in

**Comment by jordan-d-murphy** [comment]
Closed by:
[CI: speedup docstring check consecutive runs #57826](https://github.com/pandas-dev/pandas/pull/57826)
[CI: Better error control in the validation of docstrings #57879](https://github.com/pandas-dev/pandas/pull/57879) 

Opening a new issue to fix docstring errors.

**Comment by jordan-d-murphy** [comment]
Opened [DOC: Enforce Numpy Docstring Validation (Parent Issue) #58063](https://github.com/pandas-dev/pandas/issues/58063) as a parent issue for fixing docstrings based on the refactoring in code_checks.sh 

Feel free to swing by and help out! 🙂
